app/Http/Controllers/HomeController.php

```diff
@@ -42,7 +45,44 @@ public function bbcodePreview()
 
     public function getChangelog()
     {
-        return view('home.changelog');
+        $build = presence(Request::input('build'));
+
+        $changelogs = Changelog::default();
+
+        if ($build !== null) {
+            $changelogs->where('build', $build);
+        } else {
+            $from = Changelog::default()->first();
+            $changelogs->where('date', '>', Carbon::parse($from->date)->subWeeks(config('osu.changelog.recent_weeks')));
+        }
+
+        $changelogs = $changelogs
+            ->with(['gameBuild', 'user'])
+            ->orderBy('major', 'desc')
+            ->get();
+
+        if ($build === null) {
+            $changelogs = $changelogs->groupBy(function ($item) {
```

Comment:
this difference in `$changelogs` format is confusing. Might as well split by action or just put all for specific build into similar format (`$changelogs = [$changelogs]`).